pulpcore/content/handler.py

```diff
@@ -517,12 +534,13 @@ def get_latest_publication_or_version_blocking():
                     rel_path = index_path
                     headers = self.response_headers(rel_path)
                 except ObjectDoesNotExist:
-                    dir_list = await self.list_directory(None, publication, rel_path)
+                    dir_list, dates = await self.list_directory(None, publication, rel_path)
                     dir_list.update(
                         await sync_to_async(distro.content_handler_list_directory)(rel_path)
                     )
```

Comment:
Maybe allow `content_handler_list_directory` to return dates?

```suggestion
                    vals = await sync_to_async(distro.content_handler_list_directory)(rel_path)
                    if isinstance(vals, set):
                        dir_list.update(vals)
                    else:
                        dir_list.update(vals[0])
                        dates.update(vals[1])          
```